packages/docmanager-extension/src/index.tsx

```diff
@@ -857,13 +876,14 @@ function addCommands(
           if (saveInProgress.has(context)) {
             return;
           }
-
-          if (context.model.readOnly) {
-            return showDialog({
-              title: trans.__('Cannot Save'),
-              body: trans.__('Document is read-only'),
-              buttons: [Dialog.okButton()]
-            });
+          if (!context.contentsModel?.writable) {
+            if (args._luminoEvent) {
+              let type = (args._luminoEvent as ReadonlyPartialJSONObject).type;
+              if (type === 'keybinding') {
+                readonlyNotification(context.path);
+                return;
+              }
+            }
```

Comment:
Should it fall back to dialog otherwise? In practice it is unlikely to happen (only if `isWritable()` returned `true` but between the check and execution the document changed to read-only) but maybe worth ensuring that it does not enter some odd state?

Reply:
Good point. I added that showDialog back in and fixed up the logic a bit.